src/spec.c

```diff
@@ -1333,6 +1353,30 @@ void IndexSpec_Free(IndexSpec *spec) {
 
 //---------------------------------------------------------------------------------------------
 
+// Assumes this is called from the main thread with no competing threads
+// Also assumes that the spec is existing in the global dictionary, so
+// we use the global reference as our guard and access the spec dierctly.
+// This function consumes the Strong reference it gets
+void IndexSpec_RemoveFromGlobals(StrongRef ref) {
+  IndexSpec *spec = StrongRef_Get(ref);
+
+  // Remove spec from global index list
+  dictDelete(specDict_g, spec->name);
+
+  // Remove spec from global aliases list
+  if (spec->uniqueId) {
+    // If uniqueid is 0, it means the index was not initialized
+    // and is being freed now during an error.
+    IndexSpec_ClearAliases(ref);
+  }
+
+  SchemaPrefixes_RemoveSpec(ref);
+
+  // mark the spec as deleted and decrement the ref counts owned by the global dictionaries
+  StrongRef_Invalidate(ref);
+  StrongRef_Release(ref);
```

Comment:
Maybe we should write a comment here why those are the only places that should be freed and explain that other places holds a weak ref:
1. GC
2. Expire timer
3. Background indexing
4. Cursors?